openedx/edx-enterprise
# feat: add username query to filter

# Description
Adds support for searching by username to support searching by both email and username from the support tool MFE.
https://2u-internal.atlassian.net/browse/ENT-9455
# Test plan
1. add a user with a username that is distinctly different from the email address at this link: http://localhost:18000/admin/auth/user/ 
2. add that user to the an enterprise customer at http://localhost:18000/admin/enterprise/enterprisecustomer/
3. search for the user by the email and verify that you see a result
http://localhost:18000/enterprise/api/v1/enterprise-customer-support/<enterprise_customer_uuid>?user_query=<email>
4. search for the user by the username and verify that you see a result
http://localhost:18000/enterprise/api/v1/enterprise-customer-support/<enterprise_customer_uuid>?user_query=<username>
<img width="1338" alt="Screenshot 2024-09-12 at 12 37 17 AM" src="https://github.com/user-attachments/assets/d32ea30b-247a-478b-b90e-1b185b2cd1dd">
<img width="1343" alt="Screenshot 2024-09-12 at 12 37 35 AM" src="https://github.com/user-attachments/assets/9f8db203-1af4-4a9e-b67e-ff663458b6f9">




**Merge checklist:**
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

**Post merge:**
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - Trigger the '[Upgrade one Python dependency](https://github.com/openedx/edx-platform/actions/workflows/upgrade-one-python-dependency.yml)' action against master in edx-platform with new version number to generate version bump PR
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.
